### ISSUE TITLE
MySQL/MariaDB column remark in `createTable` drops column `NOT NULL` and default value

### ISSUE THREAD

**Issue by camsaul** [liquibase/liquibase#2634]
This seems to be a regression in 4.8.0; it was previously working for us on 3.6.3.

We have a changeSet like this:

```yaml
  - changeSet:
      id: 47
      author: camsaul
      validCheckSum: ANY
      changes:
        - createTable:
            tableName: collection
            remarks: 'Collections are an optional way to organize Cards and handle permissions for them.'
            columns:
              - column:
                  name: id
                  type: int
                  autoIncrement: true
                  constraints:
                    primaryKey: true
                    nullable: false
              ...
              - column:
                  name: archived
                  type: boolean
                  remarks: 'Whether this Collection has been archived and should be hidden from users.'
                  defaultValueBoolean: false
                  constraints:
                    nullable: false
```

The generated SQL looks like this::

```sql
CREATE TABLE `metabase_test`.`collection` (
  `id` INT AUTO_INCREMENT NOT NULL, 
  ...
  `archived` BIT(1) DEFAULT 0 NOT NULL COMMENT 'Whether this Collection has been archived and should be hidden from users.', 
  CONSTRAINT `PK_COLLECTION` PRIMARY KEY (`id`), 
  UNIQUE (`slug`)
) COMMENT = 'Collections are an optional way to organize Cards and handle permissions for them.';

ALTER TABLE `metabase_test`.`collection` 
MODIFY COLUMN `archived` BIT(1) 
COMMENT 'Whether this Collection has been archived and should be hidden from users.';
```

After the first statement `DESCRIBE collection;` shows

```
MariaDB [metabase_test]> describe collection;
+-------------+--------------+------+-----+---------+----------------+
| Field       | Type         | Null | Key | Default | Extra          |
+-------------+--------------+------+-----+---------+----------------+
| id          | int(11)      | NO   | PRI | NULL    | auto_increment |
...
| archived    | bit(1)       | NO   |     | b'0'    |                |
+-------------+--------------+------+-----+---------+----------------+
```

but after the second statement it is now

```
MariaDB [metabase_test]> describe collection;
+-------------+--------------+------+-----+---------+----------------+
| Field       | Type         | Null | Key | Default | Extra          |
+-------------+--------------+------+-----+---------+----------------+
| id          | int(11)      | NO   | PRI | NULL    | auto_increment |
...
| archived    | bit(1)       | YES  |     | NULL    |                |
+-------------+--------------+------+-----+---------+----------------+
```

I'm not sure why the second statement is needed at all, since the `CREATE TABLE` statement itself contains the column `COMMENT`.

Probably related: #2301

**Comment by camsaul** [comment]
So I think the problem here is that the `ALTER TABLE ... MODIFY COLUMN ...` statement needs to include `NOT NULL` and `DEFAULT` in the column definition, e.g. the second statement should look something like

```sql
ALTER TABLE `metabase_test`.`collection` 
MODIFY COLUMN `archived` BIT(1) DEFAULT 0 NOT NULL
COMMENT 'Whether this Collection has been archived and should be hidden from users.';
```

OR just skip the second statement entirely since it might not be needed.

**Comment by camsaul** [comment]
I think the line where this is compiled is here

https://github.com/liquibase/liquibase/blob/0d9c95f1a291496eed6ed55a81bd212274ff1cb5/liquibase-core/src/main/java/liquibase/sqlgenerator/core/SetColumnRemarksGenerator.java#L50

Compare to the significantly more complicated logic for `ALTER TABLE ... ADD` column statements here https://github.com/liquibase/liquibase/blob/256675bca43a089e32c940e10ed8ed7a5f9d66f6/liquibase-core/src/main/java/liquibase/sqlgenerator/core/AddColumnGenerator.java#L141-L191 (which is working correctly)... there is a lot of special casing for MySQL there

**Comment by camsaul** [comment]
RE adding column remarks statements that we don't need inside `CREATE TABLE`: that happens here

https://github.com/liquibase/liquibase/blob/d4f80c33124e850695e248437a91a0bb8e3a9e52/liquibase-core/src/main/java/liquibase/change/core/CreateTableChange.java#L148-L153 even tho `statement.addColumn()` is also passed the column remark

https://github.com/liquibase/liquibase/blob/d4f80c33124e850695e248437a91a0bb8e3a9e52/liquibase-core/src/main/java/liquibase/change/core/CreateTableChange.java#L84-L88

**Comment by camsaul** [comment]
For anyone else who runs into this: I worked around this for the time being by creating a custom subclass of `SetColumnRemarksGenerator` that no-ops for MySQL. It does mean any column remarks not added by `CREATE TABLE` won't show up, but that's definitely a better situation than losing `NOT NULL` constraints and `DEFAULT` values.

https://github.com/metabase/metabase/pull/20611/files#diff-1bf4cc0a8f85da9836ec1ffaaa41a4fb7fa66be37a0feed997589584350284bdR13-R30

and

https://github.com/metabase/metabase/pull/20611/files#diff-7b84b53f388126bbf994fbdb5b705a6eefeaf0999da00dc45a40a6559f85ac30R18-R19

**Comment by nvoxland** [comment]
This is a duplicate of #2301 and fixed by #2752. 

I'll close this version to centralize tracking
